### ISSUE TITLE
ThreadLocal integration

### ISSUE THREAD

**Issue by driventokill** [alibaba/transmittable-thread-local#130]
In the user code, `ThreadLocal` can be passed as a cross-thread transmit capability by defining it as `TransmittableThreadLocal`. However, if the user relies on a library that uses the `ThreadLocal` implementation, it cannot modify its code at this time, so it cannot be easily used in asynchronous. The ability to get context transmit in the code.

One possible solution is the **integration** transmit for `ThreadLocal`. 

Users can register the `ThreadLocal` that needs to be passed, and can provide two methods:

```java
// ThreadLocal will be transmitted after this
<T> boolean TransmittableThreadLocal.register(ThreadLocal<T> threadLocal);

// Then this ThreadLocal will not be transmitted
<T> boolean TransmittableThreadLocal.unregister(ThreadLocal<T> threadLocal);
```

----------

# `ThreadLocal`集成支持

用户代码中 `ThreadLocal` 通过定义为 `TransmittableThreadLocal` 即可获得跨线程传递能力，但是如果用户依赖的库中使用的是 `ThreadLocal` 实现，这个时候又无法修改其代码，就不能很方便的在异步化的代码中获得上下文传递的能力了。

一个可行的解决方式是开放集成的能力。

用户可以将需要传递的 `ThreadLocal` 注册进来，可以提供两个方法：

```java
// 接下来这个 ThreadLocal 会被传递
<T> boolean TransmittableThreadLocal.register(ThreadLocal<T> threadLocal);

// 接下来这个 ThreadLocal 不会被传递
<T> boolean TransmittableThreadLocal.unregister(ThreadLocal<T> threadLocal);
```

**Comment by oldratlee** [comment]
The integration transmit for `ThreadLocal` cause the **break** of **semantics** of `ThreadLocal`:

- the **Thread Confinement** from `ThreadLocal`.
    - aka. the thread-safe relied on the `ThreadLocal` does no longer established.

so mark this issue `invalid` and close.

---------------

`ThreadLocal`的集成传输 会导致 打破`ThreadLocal`的语义：

- ThreadLocal保证的线程封闭。
     - 即，依赖于`ThreadLocal`的线程安全不再成立。

鉴于上面的原因，标记这个Issue为`invalid`并Close。

**Comment by oldratlee** [comment]
released in [v2.11.0-RC3](https://github.com/alibaba/transmittable-thread-local/releases/tag/v2.11.0-RC3).
